ffi/src/lib.rs

```diff
@@ -33,6 +33,8 @@ pub mod engine_funcs;
 pub mod expressions;
 pub mod scan;
 pub mod schema;
+#[cfg(feature = "test-ffi")]
+pub mod test_ffi;
```

Comment:
I looked into this. There's no easy way to do this as far as I know. `cbindgen` [ignores](https://github.com/mozilla/cbindgen/blob/master/docs.md#ignore-annotation) all code under `#[cfg(test)]`.  

One idea is to have a separate `build.rs` that's exclusively for testing. This can include a test crate using [the bulider](https://github.com/mozilla/cbindgen/blob/master/docs.md#buildrs). 

Either way we'll have to rethink how we handle our current ffi tests imo